Remotion/Web/Development.WebTesting/HostingStrategies/DockerHosting/DockerCommandLineClient.cs

```diff
@@ -0,0 +1,238 @@
+      ArgumentUtility.CheckNotNull ("mounts", mounts);
+      ArgumentUtility.CheckNotNullOrEmpty ("imageName", imageName);
+      ArgumentUtility.CheckNotEmpty ("hostname", hostname);
+      ArgumentUtility.CheckNotEmpty ("entryPoint", entryPoint);
+      ArgumentUtility.CheckNotEmpty ("args", args);
+
+      var commandBuilder = new StringBuilder()
+          .Append ("run").Append (' ')
+          .Append ("-d").Append (' ');
+
+      if (remove)
+        commandBuilder.Append ("--rm").Append (' ');
+
+      if (ports.Any())
+      {
+        var portFlags = string.Join (" ", ports.Select (kvp => $"-p {kvp.Key}:{kvp.Value}"));
+        commandBuilder.Append (portFlags).Append (' ');
+      }
+
+      if (mounts.Any())
+      {
+        var mountFlags = string.Join (" ", mounts.Select (kvp => $@"-v ""{kvp.Key}"":""{kvp.Value}"""));
+        commandBuilder.Append (mountFlags).Append (' ');
+      }
+
+      if (entryPoint != null)
+        commandBuilder.Append ($@"--entrypoint=""{entryPoint}""").Append (' ');
+
+      if (hostname != null)
+        commandBuilder.Append ($@"--hostname ""{hostname}""").Append (' ');
+
+      commandBuilder.Append (imageName).Append (' ');
+
+      if (args != null)
+        commandBuilder.Append (args);
+
+      var command = commandBuilder.ToString();
+
+      return RunDockerCommand (command).TrimEnd ('\r', '\n');
+    }
+
+    /// <inheritdoc />
+    public bool ContainerExists (string containerName)
+    {
+      ArgumentUtility.CheckNotNullOrEmpty ("containerName", containerName);
+
+      using (var p = Process.Start (_dockerExeFullPath, $"inspect {containerName}"))
+      {
+        p.WaitForExit ((int) _commandTimeout.TotalMilliseconds);
+
+        return p.ExitCode == 0;
+      }
+    }
+
+    /// <inheritdoc />
+    public void Remove (string containerName, bool force = false)
+    {
+      ArgumentUtility.CheckNotNullOrEmpty ("containerName", containerName);
+
+      var commandBuilder = new StringBuilder()
+          .Append ("rm").Append (' ');
+
+      if (force)
+        commandBuilder.Append ("-f").Append (' ');
+
+      commandBuilder.Append (containerName);
+
+      var command = commandBuilder.ToString();
+
+      RunDockerCommand (command);
+    }
+
+    /// <inheritdoc />
+    public void Stop (string containerName)
+    {
+      ArgumentUtility.CheckNotNullOrEmpty ("containerName", containerName);
+
+      var commandBuilder = new StringBuilder()
+          .Append ("stop").Append (' ')
+          .Append (containerName);
+
+      var command = commandBuilder.ToString();
+
+      RunDockerCommand (command);
+    }
+
+    private string RunDockerCommand (string dockerCommand, string workingDirectory = null, TimeSpan? timeout = null)
+    {
+      s_log.Info ($"Running: 'docker {dockerCommand}'");
+
+      var startInfo = new ProcessStartInfo
+                      {
+                          WindowStyle = ProcessWindowStyle.Hidden,
+                          ErrorDialog = true,

```

Comment:
What does ErrorDialog = true do? could this result in a blocking popup upon error?